to_do_list/src/Components/Assignment.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import PropTypes from "prop-types";
+import EditIcon from "@material-ui/icons/Edit";
+import DeleteForeverIcon from "@material-ui/icons/DeleteForever";
+import SimpleDialog from "./SimpleDialog";
+import "../Style/Assignment.css";
+
+function Assignment(props) {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = (shouldUpdate, value) => {
+    if (shouldUpdate) {
+      props.changeItem(props.id, props.isDone, value);
+    }
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <div className="box">
+        <input
+          type="checkbox"
+          className="checkbox"
+          checked={props.isDone}
+          onChange={() => {
+            props.changeItem(props.id, !props.isDone, props.children);
+          }}
+          id={props.id}
+        />
+        <label htmlFor={props.id} className="assignmentLabel">
+          {props.children}
+        </label>
+        <EditIcon onClick={handleClickOpen} />
+        <DeleteForeverIcon onClick={() => props.delete(props.id)} />
+      </div>
+      <SimpleDialog
```

Comment:
This component shouldn't always be rendered every time you create `assignment` component.
doing it so make it only render if the open is set to true (and then you dont need to pass it down):
```js
open && <SimpleDialog />
```